### ISSUE TITLE
templates/config.json.j2: no valid json

### ISSUE THREAD

**Issue by discointheair** [ansible-collections/ansible-consul#167]
Hi 

Since https://github.com/brianshumate/ansible-consul/commit/5fdbf3fda106e174113bd49b8dda6d6c9e2a5731#diff-ca5bb9e516febbc560d531ce7f8b87e3 config.json.j2 is not valid.

At retry_join and retry_join wan the square brackets are missing.
```
-    "retry_join": [
-        {% for server in consul_join %}
-            "{{ server | ipwrap }}",
-        {% endfor %}
-        {% for server in _consul_lan_servers %}
-            "{{ hostvars[server]['consul_advertise_address'] |
-                default(hostvars[server]['consul_bind_address']) | ipwrap }}",
-        {% endfor %} ],
+    "retry_join":
+    {% for server in _consul_lan_servers %}
+    {% set consul_join = consul_join + [ hostvars[server]['consul_advertise_address'] | default(hostvars[server]['consul_bind_address']) ] %}{% if loop.last %}
+    {{ consul_join | map('ipwrap') | list | to_json }},
+    {% endif %}{% endfor %}
```
 
I would also prefer if you would like rollback this change complete so that  these lines are included: 

```
-        {% for server in consul_join %}
-            "{{ server | ipwrap }}",
-        {% endfor %}
```
Because the new code only works if  _consul_lan_servers are available. And if your playbook only includes clients this is not the case and if _consul_lan_servers exists the generated server list maybe includes redundant entries.

**Comment by brianshumate** [comment]
Hi,

Thanks for the report. Does the `to_json` not make a correctly `[]` encapsulated list then?

There is a test in `.travis.yml` that validates the json and currently it is passing so this is definitely curious.

I'll have a look.

**Comment by discointheair** [comment]
Hi brian

Thank you. 
That could be a problem at my side, because _consul_lan_servers are empty in my environment  and so to_json will not be processed. 
PS: don't now why _consul_lan_servers is empty at this moment.  The dynamic host group consul_instances I build seems to be not known in my other host group  consul_clients for which this play runs.

But makes it sense in the current template to say:
set consul_join = consul_join + [ hostvars[server]['consul_advertise_address... 
For me it seems to be better when consul_instances and _consul_lan_servers are independent from each other. 

... Send from my mobile

**Comment by coughlanio** [comment]
I encountered the same issue, and it was due to no default node role being set. I'm doing things a bit differently than recommended, as I'm using `terraform-inventory` to pull the inventory from my local terraform state.

```
---
- name: Consul Cluster
  hosts: consul
  become: yes
  become_user: root
  roles:
    - role: ansible-consul
      consul_group_name: consul
      consul_node_role: server
      consul_client_address: 0.0.0.0
```

See `consul_node_role`.

**Comment by discointheair** [comment]
Hi @coughlanio, @brianshumate 

From my point of view. There are several issues. One of them is maybe the ansible version.

@coughlanio 
There are a process order for the used variables  consul_instances, _consul_lan_servers,  consul_servers which are later relevant that this code in config.json.j2 works:
```
  {% for server in _consul_lan_servers %}
    {% set consul_join = consul_join + [ hostvars[server]['consul_advertise_address'] | default(hostvars[server]['consul_bind_address']) ] %}
```
I think this is the minimal set the this works (for me consul_instances is only a dummy group to make this role runnable and setup consul clients):
```
- name: "Create consul server group"
  add_host:
    name: "{{ item }}"
    groups: "consul_instances"
    consul_node_role: "server"
    consul_advertise_address: "{{ item }}"
    consul_datacenter: "{{ _datacenter }}"
  with_items:
    - "1.1.1.1"
```
@brianshumate 

To my problem here:
```
{% for server in _consul_lan_servers %}
   {% set consul_join = consul_join + [ hostvars[server]['consul_advertise_address'] | default(hostvars[server]['consul_bind_address']) ] %}{% if loop.last %}
  {{ consul_join | map('ipwrap') | list | to_json }},
  {% endif %}{% endfor %}
```

"set consul_join = consul_join +..." do not works in the loop because every allocation is only valid in the current run-through. that means that the result of   "{{ consul_join | map('ipwrap') | list | to_json }}," is every time consul_join[] + the last entry in hostvars[server]['consul_bind_address']. Maybe its a problem with the ansible version. I use: ansible 2.5.2. Which version do you use?

In total the change still makes no sense for me because of the "condition" in  "{% for server in _consul_lan_servers %}" I need do define consul_instances with all consul servers and it makes no sense to use consul_join then.

**Comment by brianshumate** [comment]
I think this is now resolved if on Ansible 2.5+ after this fix in #176
